app/services/merge_collection.rb

```diff
@@ -0,0 +1,228 @@
+# frozen_string_literal: true
+
+class MergeCollection
+  class << self
+    def call(uuid, delete_collection: true)
+      instance = new(uuid, delete_collection)
+      instance.save
+      instance
+    end
+  end
+
+  attr_reader :uuid,
+              :delete_collection,
+              :errors
+
+  def initialize(uuid, delete_collection)
+    @uuid = uuid
+    @delete_collection = delete_collection
+    @errors = []
+    @successful = false # updated by #save
+  end
+
+  def save
+    return false if collection.nil?
+    return false unless valid?
+
+    ActiveRecord::Base.transaction do
+      PaperTrail.request(whodunnit: paper_trail_user) do
+        new_work = merge_collection
+
+        if !new_work.valid?
+          new_work.errors.full_messages.each { |err| errors << err }
+          return false
+        end
+
+        new_work.save!
+
+        new_work.update_index
+      end
+
+      if delete_collection?
+        collection.destroy!
+
+        collection.works.each do |work|
+          work.versions.each do |version|
+            DestroyWorkVersion.call(version, force: true)
+          end
+        end
```

Comment:
@nicktiberi this potentially could move to its own small method as well, but totally up to you, I think the PR looks good overall.